src/org/folio/rest_v2/Eureka.groovy

```diff
@@ -0,0 +1,95 @@
+package org.folio.rest_v2
+
+import org.folio.models.EurekaTenant
+import org.folio.utilities.RequestException
+
+class Eureka extends Authorization {
+
+  /**
+   * EurekaTenant object contains Master Tenant configuration for Eureka.
+   * @attribute 'tenantId'     Default: "master"
+   * @attribute 'clientId'     Default: "folio-backend-admin-client"
+   */
+  public EurekaTenant masterTenant
+
+  Eureka(Object context, String eurekaDomain, EurekaTenant masterTenant, boolean debug = false) {
+    super(context, eurekaDomain, debug)
+    this.masterTenant = masterTenant
+  }
+
+  void createTenant(EurekaTenant tenant) {
+    if (isTenantExist(tenant.tenantId)) {
+      logger.warning("Tenant ${tenant.tenantId} already exists!")
+      return
+    }
+
+    String url = generateUrl("/tenants")  // Tenant Manager URL
+    Map<String, String> headers = getMasterHeaders()
+    Map body = [
+      name: tenant.tenantId,
+      description: tenant.tenantDescription
+    ]
+
+    logger.info("Creating tenant ${tenant.tenantId}...")
+
+    restClient.post(url, body, headers)
+
+    logger.info("Tenant (${tenant.tenantId}) successfully created")
+  }
+
+  boolean isTenantExist(String tenantId) {
+    String url = generateUrl("/tenants/${tenantId}")
+    Map<String, String> headers = getMasterHeaders()
+
+    try {
+      restClient.get(url, headers)
+      logger.info("Tenant ${tenantId} exists")
+      return true
+    } catch (RequestException e) {
+      if (e.statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+        logger.info("Tenant ${tenantId} is not exists")
+        return false
+      } else {
+        throw new RequestException("Can not able to check tenant ${tenantId} existence: ${e.getMessage()}", e.statusCode)
```

Comment:
be careful with throwing an exception, that will abort whole pipeline execution...

Reply:
Code is re-written to avoid throwing exception in this place.